src/class/Command/join.cpp

```diff
@@ -7,10 +7,15 @@ static void handler(const args_t &args, Client &client, Server &server)
 {
 	size_t args_size = args.size();
 
-	// if the only argument to /join 0
 	if (args_size == 2 && args[1] == "0")
-		// TODO close_all_chanels() -> would be easier to implement when the /PART command will be
+	{
+		if (client.get_channels_count() != 0)
+		{
+			std::string reason = client.get_nickname();
+			client.close_all_channels(reason);
+		}
 		return ;
+	}
```

Comment:
La verification ne me semble pas necessaire. Dans tous les cas tu vas les parcourir ensuite non ?

```suggestion
	{
	        std::string reason = client.get_nickname();
	        client.close_all_channels(reason);
		return ;
	}
```